cdap-common/src/main/java/co/cask/cdap/common/MethodNotAllowed.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common;
+
+/**
+ * Thrown when some method is not allowed.
+ */
+public class MethodNotAllowed extends Exception {
+
+  public MethodNotAllowed(String message) {
```

Comment:
should probably store the method that is not allowed, and then construct the message, for consistency?